asteroid/engine/gan_system.py

```diff
@@ -0,0 +1,161 @@
+from collections import OrderedDict
+import soundfile as sf
+import pytorch_lightning as pl
+import torch
+import os
+
+
+class GanSystem(pl.LightningModule):
+
+    def __init__(self, discriminator, generator, opt_d, opt_g,
+                 scheduler_d, scheduler_g, discriminator_loss, generator_loss,
+                 validation_loss, train_loader, val_loader=None, conf=None):
+
+        super(GanSystem, self).__init__()
+        self.discriminator = discriminator
+        self.generator = generator
+        self.opt_d = opt_d
+        self.opt_g = opt_g
+        self.d_loss = discriminator_loss
+        self.g_loss = generator_loss
+        self.validation_loss = validation_loss
+        self.train_loader = train_loader
+        self.val_loader = val_loader
+        self.scheduler_d = scheduler_d
+        self.scheduler_g = scheduler_g
+        self.conf = conf
+
+    def forward(self, z):
+        return self.generator(z)
```

Comment:
I would have *args, **kwargs here. It would be more generic for conditional GANs for example

Reply:
I don't understand how we can pass more arguments to the generator anyway since the call is done in `gan_system` and is never to be changed by the user or am I missing something ?

Reply:
Different GANs will have different `nn.Module` as generator. 
```suggestion
    def forward(self, *args, **kwargs):
        return self.generator(*args, **kwargs)
```
This is just more generic.